apps/ios/LiveVideo/LiveVideo/Twilio/Sync/SyncUsersStore.swift

```diff
@@ -5,25 +5,24 @@
 import TwilioSyncClient
 import Combine
 
-class RaisedHandsStore: NSObject, SyncStoring, ObservableObject {
-    struct RaisedHand: Identifiable {
-        let userIdentity: String
-        var id: String { userIdentity }
-        
+/// Reusable store that fetches users from a sync map.
+///
+/// Set `uniqueName` to specify the name of the sync map to fetch users from.
+class SyncUsersStore: NSObject, SyncStoring {
+    struct User: Identifiable {
+        let identity: String
+        var id: String { identity }
+     
         init(mapItem: TWSMapItem) {
-            userIdentity = mapItem.key
+            identity = mapItem.key
         }
     }
 
-    @Published var raisedHands: [RaisedHand] = []
-    @Published var haveNew = false
-    private var newRaisedHands: [RaisedHand] = [] {
-        didSet {
-            haveNew = !newRaisedHands.isEmpty
-        }
-    }
+    let userAddedPublisher = PassthroughSubject<User, Never>()
+    let userRemovedPublisher = PassthroughSubject<User, Never>()
     var uniqueName: String!
     var errorHandler: ((Error) -> Void)?
+    private(set) var users: [User] = []
```

Comment:
Nit, but perhaps would a Set would work better here because the order of users is not useful in this layer.

Reply:
Good question but I think it's probably best to keep an array. SwiftUI needs an array. And sync provides an array. And so it would make for a lot of converting between set and array.